### ISSUE TITLE
VM disk file_format always gets set to "raw"

### ISSUE THREAD

**Issue by 1-cameron** [bpg/terraform-provider-proxmox#210]
**Describe the bug**
I am unable to deploy a VM with a `qcow2` or `vmdk` file format disk. 

**To Reproduce**
1. Define a proxmox_virtual_environment_vm resource in terraform with a disk that has the file_format set to `qcow2` or `vmdk`
2. terraform apply

**Expected behavior**
The hard disk attached matches the definition in the config.

**Screenshots**
The plan and apply output the following:
```
+ disk {
  + datastore_id = "local"
  + file_format  = "qcow2"
  + interface    = "scsi0"
  + iothread     = false
  + size         = 32
  + ssd          = false
}
```

But the provisioned VM has a raw formatted hard disk attached:
![image](https://user-images.githubusercontent.com/68611194/212918248-ecd771ad-ce5e-4c09-868d-1f2f4ded4fde.png)

**Additional context**
I'm using v0.9.1 of the provider and Proxmox VE 7.3-3

**Comment by bpg** [comment]
Hey @1-cameron, would you be able create a `qcow2` disk on this storage in the PVE UI?

From the docs https://pve.proxmox.com/pve-docs/chapter-qm.html#qm_hard_disk:
> The choice of a storage type will determine the format of the hard disk image. Storages which present block devices (LVM, ZFS, Ceph) will require the raw disk image format, whereas files based storages (Ext4, NFS, CIFS, GlusterFS) will let you to choose either the raw disk image format or the QEMU image format.

**Comment by 1-cameron** [comment]
I'm able to create the disks manually in the UI in all formats on that storage with no issue.

However, I've just noticed that the disks I've been provisioning are actually larger than the space available on the storage. I've since provisioned disks with the expected file_format on larger storage (they're the same type).

Problem storage:
![image](https://user-images.githubusercontent.com/68611194/213679834-f5f3f5de-70f6-42a3-a147-2a7a12424594.png)

Working storage:
![image](https://user-images.githubusercontent.com/68611194/213679917-c75e805c-5cac-4299-ab53-806084a64a3c.png)

**Comment by yuqo2450** [comment]
Hey there,

I am having the same issue.
I even tried to set it to _raw_, but even then _terraform apply_ plans to recreate the vm.

**Comment by 1-cameron** [comment]
> Hey @1-cameron, thanks, I'm planning to re-work the VM disk provisioning to address #187, #188, and #203 -- they all are related to the same [piece of code](https://github.com/bpg/terraform-provider-proxmox/issues/187#issuecomment-1363594575) that is really hard to maintain. I'll add you use case to the test suite.

Thanks, I really appreciate the work you're doing on this provider!

**Comment by bpg** [comment]
Should be fixed by #241
